crates/services/gas_price_service/src/lib.rs

```diff
@@ -25,6 +31,8 @@ pub struct GasPriceService<A, U> {
     next_block_algorithm: SharedGasPriceAlgo<A>,
     /// The code that is run to update your specific algorithm
     update_algorithm: U,
+    // The da block cost sub service
+    da_block_costs_task_handle: ServiceRunner<DaBlockCostsService<DummyDaBlockCosts>>,
```

Comment:
I think this handle can just be on the adapter. 

```
#[derive(Clone)]
pub struct DaBlockCostsProvider {
    service_handle: ServiceRunner<DaBlockCostsService<DummyDaBlockCosts>>,
    receiver: Arc<Mutex<mpsc::Receiver<DaBlockCosts>>>,
}
```
Why not?

Reply:
The primary reason not to place the `service_handle` directly on the adapter is that it violates separation of concerns and single responsibility principle. The adapter’s role should be strictly focused on fetching and providing the da block costs. wdyt?

Reply:
I don't think it's a violation of either of those.

It would be violation of SoC if we had the client also making calls for the L2block info. Those things can vary independently, and it would be wrong to couple them. The concern here is getting info from the committer. Our _implementation_ is having a service that runs in another thread that eagerly buffers the data for us. If we were doing it lazily--without the service thread--everything would be in the adapter as well, and I assume you wouldn't have any issue with that.

The responsibility here is getting the costs from DA. It's not a misdirection at all for something called `DaBlockCostsProvider` to have a handle to task that is providing us with the latest `DaBlockCosts`. 

I definitely think the `GasPriceService` is the _wrong_ place for it. It has no concept of L2 or DA at that level. That's the job of our specific `update_algorithm` (not the best name, mb)

The only question I have about if this is the "right" pace to hold the handle is how we want to handle errors from the service.